spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/SpannerRepositoryFactoryTests.java

```diff
@@ -0,0 +1,98 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.repository;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.cloud.gcp.data.spanner.core.SpannerOperations;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerColumn;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerMappingContext;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerTable;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mapping.MappingException;
+import org.springframework.data.repository.core.EntityInformation;
+import org.springframework.data.repository.core.RepositoryInformation;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Chengyuan Zhao
+ */
+@RunWith(SpringRunner.class)
+public class SpannerRepositoryFactoryTests {
+
+	private SpannerRepositoryFactory spannerRepositoryFactory;
+
+	private SpannerMappingContext spannerMappingContext;
+
+	private SpannerOperations spannerOperations;
+
+	@Before
+	public void setUp() {
+		this.spannerMappingContext = new SpannerMappingContext();
+		this.spannerOperations = mock(SpannerOperations.class);
+		this.spannerRepositoryFactory = new SpannerRepositoryFactory(
+				this.spannerMappingContext, this.spannerOperations);
+	}
+
+	@Test
+	public void getEntityInformationTest() {
+		EntityInformation entityInformation = this.spannerRepositoryFactory
+				.getEntityInformation(TestEntity.class);
+		assertEquals(TestEntity.class, entityInformation.getJavaType());
+		assertEquals(String.class, entityInformation.getIdType());
+	}
+
+	@Test(expected = MappingException.class)
+	public void getEntityInformationNotAvailableTest() {
+		SpannerRepositoryFactory factory = new SpannerRepositoryFactory(
+				mock(SpannerMappingContext.class), this.spannerOperations);
+		factory.getEntityInformation(TestEntity.class);
+	}
+
+	@Test
+	public void getTargetRepositoryTest() {
+		RepositoryInformation repoInfo = mock(RepositoryInformation.class);
+		when(repoInfo.getRepositoryBaseClass())
+				.thenReturn((Class) SpannerRepositoryImpl.class);
+		when(repoInfo.getDomainType()).thenReturn((Class) TestEntity.class);
+		Object repo = this.spannerRepositoryFactory.getTargetRepository(repoInfo);
+		assertTrue(SpannerRepository.class.isAssignableFrom(repo.getClass()));
```

Comment:
I'm not sure why you need to use `isAssignableFrom`? It seems that we should expect the exact class returned from the repoInfo

Reply:
I think all we need to know is that it's returning a SpannerRepository , not any specific implementation.

Reply:
We'll need the actual equality on `SpannerRepositoryImpl.class`, based on the instructions in the [RepositoryFactorySupport javadoc](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/core/support/RepositoryFactorySupport.html#getRepositoryBaseClass-org.springframework.data.repository.core.RepositoryMetadata-)

````
	/**
	 * Returns the base class backing the actual repository instance. Make sure
	 * {@link #getTargetRepository(RepositoryInformation)} returns an instance of this class.
	 *
	 * @param metadata
	 * @return
	 */
	protected abstract Class<?> getRepositoryBaseClass(RepositoryMetadata metadata);
````